File: backend/TravelWeb-API/TravelWeb-API/DTO/ActivityDTO/RouteOptionDTO.cs
namespace TravelWeb_API.DTO.ActivityDTO
{
    public class RouteOptionDTO
    {
        public string Summary { get; set; } = "";
        public string DistanceText { get; set; } = "";
        public string DurationText { get; set; } = "";
        public int DistanceMeters { get; set; }
        public string EncodedPolyline { get; set; } = "";
        public List<string> RouteLabels { get; set; } = new();
        public List<RouteStepDTO> Steps { get; set; } = new();
    }
}

File: frontend/TravelFrontEnd/src/app/Activity/Interface/routeOptionDto.ts
import { routeStepDto } from "./routeStepDto";

export interface RouteOptionDto {
  summary: string;
  distanceText: string;
  durationText: string;
  distanceMeters: number;
  encodedPolyline: string;
  routeLabels: string[];
  steps: routeStepDto[];
}
